core/foundation/CMakeLists.txt

```diff
@@ -23,24 +23,41 @@ set(Foundation_dict_headers
   PARENT_SCOPE
 )
 
-set(sources
+set(FOUNDATION_SOURCES
   src/FoundationUtils.cxx
   src/RConversionRuleParser.cxx
   src/TClassEdit.cxx
   src/TError.cxx
 )
 
-ROOT_OBJECT_LIBRARY(Foundation ${sources})
+if(CMAKE_CXX_STANDARD GREATER 11)
+  list(APPEND Foundation_dict_headers ROOT/RLogger.hxx)
+  list(APPEND FOUNDATION_SOURCES v7/src/RLogger.cxx)
+  set(FOUNDATION_V7_INC ${CMAKE_SOURCE_DIR}/core/foundation/v7/inc)
+  # TLogger.hxx may be used without root7 flag, but is placed in v7/inc/,
```

Comment:
Shouldn't we move TLogger.cxx out of the v7/inc directory since we moving it anyway?

Reply:
Good catch! Done!